src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -394,6 +395,34 @@ public DateTime LastSaved
         }
 
         /// <summary>
+        /// gathers the direct workspace dependencies of this workspace.
+        /// </summary>
+        /// <returns> a list of workspace IDs in GUID form</returns>
+        public HashSet<Guid> Dependencies
+        {
+            get {
+                dependencies.Clear();
+                //if the workspace is a main workspace then find all functions and their dependencies
+                if (this is HomeWorkspaceModel)
+                {
+                    foreach (var node in this.Nodes.OfType<Function>())
+                    {
+                        dependencies.Add(node.GUID);
+                    }
+                }
+                //else the workspace is a customnode - and we can add the dependencies directly
+                else
+                {
+                    foreach (var id in (this as CustomNodeWorkspaceModel).CustomNodeDefinition.DirectDependencies.Select(x => x.FunctionId))
```

Comment:
You might be able to turn linq results into hashset directly instead of inserting one by one yourself. https://stackoverflow.com/questions/3471899/how-to-convert-linq-results-to-hashset-or-hashedset